main/solution/post-deployment/config/infra/cloudformation.yml

```diff
@@ -465,6 +476,66 @@ Resources:
                 Resource:
                   - 'arn:aws:elasticloadbalancing:*:*:targetgroup/analysis-*'
 
+  # =============================================================================================
+  # Cognito PreSignUp trigger
+  # =============================================================================================
+
+  PreSignUpLambdaRole:
+    Condition: PreSignUpNativePoolUsers
+    Type: 'AWS::IAM::Role'
+    Properties:
+      RoleName: !Sub '${self:custom.settings.namespace}-Cognito-PreSignUp-Role'
+      Path: '/'
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: 'Allow'
+            Principal:
+              Service:
+                - 'lambda.amazonaws.com'
+            Action:
+              - 'sts:AssumeRole'
+      Policies:
+        - PolicyName: cloudwatch-access
+          PolicyDocument:
+            Statement:
+              - Effect: 'Allow'
+                Action: logs:CreateLogGroup
+                Resource: !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:*'
+              - Effect: 'Allow'
+                Action:
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/${self:custom.settings.namespace}-Cognito-PreSignUp:*'
+
+  # Lambda that automatically confirms newly signed up Cognito users based on pre-defined logic
+  PreSignUpLambda:
+    Condition: PreSignUpNativePoolUsers
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Description: PreSignUp lambda to auto-confirm newly signed up users in Cognito
+      Runtime: nodejs14.x
+      Handler: index.handler
+      FunctionName: !Sub '${self:custom.settings.namespace}-Cognito-PreSignUp'
+      Role: !GetAtt PreSignUpLambdaRole.Arn
+      Code:
+        ZipFile: |
+          exports.handler = (event, context, callback) => {
+          event.response.autoConfirmUser = true;
+          const emailRegex = /^\w+([\.+-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
```

Comment:
Do we want to add a comment referencing the `createUser` regex and vise versa, so if anyone ever change that regex they can update both places?